### PR TITLE
Add "static constructor" pattern

### PR DESCRIPTION
**Pull request by SeanTAllen**

### CODE REVIEW

File: docs/creation/static-constructor.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+---
+hide:
+  - toc
+---
+
+# Static Constructor
+
+## Problem
+
+You want to construct an object or return a meaningful error message. Unfortunately, in Pony there's no way to do that. Pony constructors always return an initialized instance of their class unless, the constructor is partial in which case nothing is return as we jump to the nearest error handler.
```
Comment (rhagenson) [inline docs/creation/static-constructor.md:10]: As written these first two sentences read like there is no way in Pony to "construct an object or return a meaningful error message" -- which is not the case.

Comment (rhagenson) [inline docs/creation/static-constructor.md:10]:
```suggestion
You want to construct an object or return a meaningful error message. Unfortunately, in Pony there's no way to do that. Pony constructors always return an initialized instance of their class unless, the constructor is partial in which case nothing is returned as we jump to the nearest error handler.
```

File: docs/creation/static-constructor.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+```pony
+actor Main
+  new create(env: Env) =>
+    match FooConstructor(true)
+    | let f: Foo =>
+      // ToDo: do something with Foo
+      None
+    | let e: Error =>
+      env.err.print(e.msg)
```
Comment (rhagenson) [inline docs/creation/static-constructor.md:82]:
```suggestion
actor Main
  new create(env: Env) =>
    match FooConstructor(true)
    | let f: Foo =>
      // ToDo: do something with Foo
      None
    | let e: Error =>
      env.err.print(e.msg)
    end
```

**Review by rhagenson — COMMENTED** [review]
Just a few edits. The problem statement needs additional information.

File: docs/creation/static-constructor.md (hunk 1 above)
Comment (rhagenson) [reply docs/creation/static-constructor.md]: The additions you made addressed my concern.

**Review by jemc — APPROVED** [review]

**Pull request merged**